hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/amrmproxy/FederationInterceptor.java

```diff
@@ -770,7 +771,7 @@ public FinishApplicationMasterResponse finishApplicationMaster(
 
     if (failedToUnRegister) {
       homeResponse.setIsUnregistered(false);
-    } else {
+    } else if (request.getFinalApplicationStatus().equals(FinalApplicationStatus.SUCCEEDED)) {
```

Comment:
To prevent null pointers, it is usually to do it the other way.
```
} else if (FinalApplicationStatus.SUCCEEDED.equals(request.getFinalApplicationStatus()) {
```
At the same time, we need to check request is not null before.

Reply:
Thank you very much for helping to review the code, I will modify the code.